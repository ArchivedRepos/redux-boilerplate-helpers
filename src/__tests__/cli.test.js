/* eslint-env jest */

import webpack from 'webpack';
import fs from 'fs';
import { join } from 'path';
import mkdirp from 'mkdirp';
import del from 'del';
import run from '../testUtils';
import webpackConfig from '../../webpack.config.babel';

describe('cli', () => {
  beforeAll(done => {
    webpack(webpackConfig[1], done);
  });

  it('adds actions to existing files in dry run', () => {
    const cmd = run(join(__dirname, '../__testfixtures__'), [
      '--dir',
      './addToExisting',
      'oneAction',
      'twoAction',
      '--dry',
      '--single-quote',
      '--trailing-comma',
      'all',
    ]);
    expect(cmd.stderr).toEqual('');
    expect(cmd.stdout).toMatchSnapshot();
  });

  it('creates files if they are not present', () => {
    const dir = join(__dirname, '../__testfixtures__/blank');
    mkdirp.sync(dir);
    const cmd = run(join(__dirname, '../__testfixtures__'), [
      '--dir',
      './blank',
      'oneAction',
      'twoAction',
      '--single-quote',
      '--trailing-comma',
      'all',
    ]);
    const constResult = fs.readFileSync(join(dir, 'constants.js'), 'utf8');
    const actionsResult = fs.readFileSync(join(dir, 'actions.js'), 'utf8');
    const reducerResult = fs.readFileSync(join(dir, 'reducer.js'), 'utf8');
    expect(cmd.stderr).toEqual('');
    expect(constResult).toMatchSnapshot();
    expect(actionsResult).toMatchSnapshot();
    expect(reducerResult).toMatchSnapshot();
    del.sync([join(__dirname, '../__testfixtures__/blank/**')]);
  });
});
