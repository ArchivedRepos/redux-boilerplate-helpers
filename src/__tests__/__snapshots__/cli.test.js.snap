// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli adds actions to existing files in dry run 1`] = `
"==== addToExisting/constants.js ====
export const TEST_ACTION = '__testfixtures__/TEST_ACTION';
export const ANOTHER_ACTION = '__testfixtures__/ANOTHER_ACTION';
export const ONE_ACTION = 'addToExisting/ONE_ACTION';
export const TWO_ACTION = 'addToExisting/TWO_ACTION';

==== addToExisting/actions.js ====
import { createAction } from 'redux-boilerplate-helpers';
import {
  TEST_ACTION,
  ANOTHER_ACTION,
  ONE_ACTION,
  TWO_ACTION,
} from './constants';

export const testAction = createAction(TEST_ACTION);
export const anotherAction = createAction(ANOTHER_ACTION);
export const oneAction = createAction(ONE_ACTION);
export const twoAction = createAction(TWO_ACTION);

==== addToExisting/reducer.js ====
import {
  TEST_ACTION,
  ANOTHER_ACTION,
  ONE_ACTION,
  TWO_ACTION,
} from './constants';

const initialState = {};

function testReducer(state = initialState, action) {
  switch (action.type) {
    case TEST_ACTION:
      return { ...state, test: 10 };
    case ANOTHER_ACTION:
      return { ...state, another: 'test' };
    case ONE_ACTION:
      return {
        ...state,
      };
    case TWO_ACTION:
      return {
        ...state,
      };
    default:
      return state;
  }
}

export default testReducer;

"
`;

exports[`cli creates files if they are not present 1`] = `
"export const ONE_ACTION = 'blank/ONE_ACTION';
export const TWO_ACTION = 'blank/TWO_ACTION';
"
`;

exports[`cli creates files if they are not present 2`] = `
"import { createAction } from 'redux-boilerplate-helpers';
import { ONE_ACTION, TWO_ACTION } from './constants';
export const oneAction = createAction(ONE_ACTION);
export const twoAction = createAction(TWO_ACTION);
"
`;

exports[`cli creates files if they are not present 3`] = `
"import { ONE_ACTION, TWO_ACTION } from './constants';
const initialState = {};

function reducer(state = initialState, action) {
  switch (action.type) {
    case ONE_ACTION:
      return {
        ...state,
      };
    case TWO_ACTION:
      return {
        ...state,
      };
    default:
      return state;
  }
}

export default reducer;
"
`;
